opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -227,6 +227,60 @@ namespace WellGroupHelpers
         return gefac * rate;
     }
 
+    void updateGuideRatesForInjectionGroups(const Group& group,
+                                            const Schedule& schedule,
+                                            const SummaryState& summaryState,
+                                            const Opm::PhaseUsage& pu,
+                                            const int reportStepIdx,
+                                            const WellStateFullyImplicitBlackoil& wellState,
+                                            GuideRate* guideRate)
+    {
+        for (const std::string& groupName : group.groups()) {
+            const Group& groupTmp = schedule.getGroup(groupName, reportStepIdx);
+            updateGuideRatesForInjectionGroups(
+                        groupTmp, schedule, summaryState, pu, reportStepIdx, wellState, guideRate);
+        }
+        const Phase all[] = {Phase::WATER, Phase::OIL, Phase::GAS};
+        for (Phase phase : all) {
+
+            if(!group.hasInjectionControl(phase))
+                continue;
+
+            double guideRateValue = 0.0;
+            const auto& controls = group.injectionControls(phase, summaryState);
+            switch (controls.guide_rate_def){
+            case Group::GuideRateInjTarget::RATE:
+                break;
+            case Group::GuideRateInjTarget::VOID:
+            {
+                guideRateValue = wellState.currentInjectionVREPRates(group.name());
+                break;
+            }
+            case Group::GuideRateInjTarget::NETV:
+            {
+                guideRateValue = wellState.currentInjectionVREPRates(group.name());
+                const std::vector<double>& injRES = wellState.currentInjectionGroupReservoirRates(group.name());
+                if (phase != Phase::OIL && pu.phase_used[BlackoilPhases::Liquid])
+                    guideRateValue -= injRES[pu.phase_pos[BlackoilPhases::Liquid]];
+                if (phase != Phase::GAS && pu.phase_used[BlackoilPhases::Vapour])
+                    guideRateValue -= injRES[pu.phase_pos[BlackoilPhases::Vapour]];
+                if (phase != Phase::WATER && pu.phase_used[BlackoilPhases::Aqua])
+                    guideRateValue -= injRES[pu.phase_pos[BlackoilPhases::Aqua]];
+                break;
+            }
+            case Group::GuideRateInjTarget::RESV:
+                OPM_THROW(std::runtime_error, "GUIDE PHASE RESV not implemented. Group " + group.name());
+            case Group::GuideRateInjTarget::POTN:
+                break;
+            case Group::GuideRateInjTarget::NO_GUIDE_RATE:
+                break;
+            default:
+                assert(false);
+            }
+            guideRate->injectionGroupCompute(group.name(), phase, reportStepIdx, guideRateValue);
```

Comment:
The group efficiency for the production group is applied at [WellGroupHelpers.hpp#L156-L162](https://github.com/OPM/opm-simulators/blob/205c4839192d2aa5ced7d6cb88ccede78f246dcd/opm/simulators/wells/WellGroupHelpers.hpp#L156-L162) when we compute the guide rates for production. Shouldn't this also be done before this line. 
But I might be missing something.

Reply:
From my understanding it should already be accounted for here https://github.com/OPM/opm-simulators/blob/c29b61ff999969c0964cd6e39131661c7c3ea123/opm/simulators/wells/WellGroupHelpers.cpp#L162